backstory/src/test/java/com/google/sps/perspective/PerspectiveManagerTest.java

```diff
@@ -0,0 +1,174 @@
+      StoryDecision actual = manager.generateDecision("foo");
+      StoryDecision expected = new StoryDecision("foo");
+
+      Assert.assertEquals(expected, actual);
+    } catch (NoAppropriateStoryException exception) {
+      Assert.fail(exception.toString());
+    }
+  }
+
+  /**
+   * Call generateDecision() with an inappropriate story 
+   * (toxicity manually set above 70%) to ensure that it will throw
+   * a NoAppropriateStoryException
+   */
+  @Test 
+  public void checkInappropriateStory() {
+    final float TOO_TOXIC = .71f;
+    desiredScores.put(AttributeType.TOXICITY, TOO_TOXIC);
+    api = createMockAPI(desiredScores);
+
+    PerspectiveManager manager = new PerspectiveManager(api);
+    try {
+      manager.generateDecision("foo");
+    } catch (NoAppropriateStoryException exception) {
+      return;
+    }
+
+    Assert.fail("NoAppropriateStoryException should have been thrown");
+  }
+
+  /** 
+   * Returns a mock version of the PerspectiveAPI which will return the scores
+   * passed in through desiredScores or which returns -1 when there is no desired score
+   * (i.e. the map does not contain that attribute type as a key).
+   *
+   * @param desiredScores a map of the scores you want the mockAPI to return 
+   *    (if you don't set any, the mock version of this API will always return -1)
+   * @return the mocked version of the PerspectiveAPI
+   */
+  private static PerspectiveAPI createMockAPI(Map<AttributeType, Float> desiredScores) {
+    // instantiate all the necessary mock objects
+    PerspectiveAPI mockAPI = mock(PerspectiveAPI.class);
+    AnalyzeCommentResponse mockResponse = mock(AnalyzeCommentResponse.class);
+    AttributeScore mockAttributeScore = mock(AttributeScore.class);
+    Score mockScore = mock(Score.class);
+
+    // when analyze is called for our mockAPI, return the mockResponse
+    when(mockAPI.analyze(any(AnalyzeCommentRequest.class))).thenReturn(mockResponse);
+
+    // when getAttributeScore is called we need to be able to grab the type it is looking for
+    // then, check if this AttributeType is in the map and if it is, grab its value, and
+    // set up the next few calls so that this value will be returned. if it isn't in the map,
+    // set up the next few calls so -1 will be returned (-1 signifies no particular value was desired).
+    when(mockResponse.getAttributeScore(any(AttributeType.class))).thenAnswer(new Answer() {
+      public Object answer(InvocationOnMock invocation) {
+        Object[] args = invocation.getArguments();
+        Object mock = invocation.getMock();
+
+        float fakeScore = -1;
+        AttributeType type = (AttributeType) args[0];
+
+        if (desiredScores.containsKey(type)) {
+          fakeScore = desiredScores.get(type);
+        }
+
+        when(mockAttributeScore.getSummaryScore()).thenReturn(mockScore);
+        when(mockScore.getValue()).thenReturn(fakeScore);
+
+        return mockAttributeScore;
+      }
+    });
+
+    return mockAPI;
+  }
```

Comment:
Can this be consolidated into some helper class, to avoid duplication with the PerspectiveAPIClientTests?

Reply:
I created a `MockPerspectiveAPIFactory` in the text folder that will manufacture instances like this for my classes.

Reply:
Done